docs/source/typevars.py

```diff
@@ -0,0 +1,103 @@
+"""Transform references to typevars to avoid missing reference errors.
+
+See https://github.com/sphinx-doc/sphinx/issues/7722 also.
+"""
+from __future__ import annotations
+
+import re
+
+from pathlib import Path
+
+from sphinx.addnodes import pending_xref, Element
+from sphinx.application import Sphinx
+from sphinx.environment import BuildEnvironment
+from sphinx.errors import NoUri
+
+
+def identify_typevars() -> None:
+    """Record all typevars in trio."""
+    trio_folder = Path(__file__, "..", "..", "..", "trio").resolve()
+    print(trio_folder)
+    for filename in trio_folder.rglob("*.py"):
+        with open(filename, encoding="utf8") as f:
+            for line in f:
+                # A simple regex should be sufficient to find them all, no need to actually parse.
+                match = re.search(
+                    r"^\s*(\w+)\s*=\s*(TypeVar|TypeVarTuple|ParamSpec)\(",
+                    line,
+                )
+                if match is not None:
+                    relative = "trio" / filename.relative_to(trio_folder)
+                    relative = relative.with_suffix("")
+                    if relative.name == "__init__":  # Package, remove.
+                        relative = relative.parent
+                    kind = match.group(2)
+                    name = match.group(1)
+                    typevars_qualified[f'{".".join(relative.parts)}.{name}'] = kind
+                    existing = typevars_named.setdefault(name, kind)
+                    if existing != kind:
+                        print("Mismatch: {} = {}, {}", name, existing, kind)
+
+
+# All our typevars, so we can suppress reference errors for them.
+typevars_qualified = {
+    "_os.PathLike": "typing.TypeVar",
```

Comment:
Why?

Reply:
Seems to not be necessary, thought it was at some point to stop this errorring.